tests/tensorflow/nightly/vision-retinanet.libsonnet

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+local experimental = import '../experimental.libsonnet';
+local common = import 'common.libsonnet';
+local mixins = import 'templates/mixins.libsonnet';
+local timeouts = import 'templates/timeouts.libsonnet';
+local tpus = import 'templates/tpus.libsonnet';
+local utils = import 'templates/utils.libsonnet';
+
+{
+  local coco = {
+    scriptConfig+: {
+      trainFilePattern: '$(COCO_DIR)/train*',
+      evalFilePattern: '$(COCO_DIR)/val*',
+    },
+  },
+  local retinanet = common.TfVisionTest + coco {
+    modelName: 'vision-retinanet',
+    scriptConfig+: {
+      experiment: 'retinanet_resnetfpn_coco',
+      paramsOverride+: {
+        task+: {
+          annotation_file: '$(COCO_DIR)/instances_val2017.json',
+        },
+      },
+    },
+  },
+  local functional = common.Functional {
+    scriptConfig+: {
+      paramsOverride+: {
+        trainer+: {
+          train_steps: 200,
+          validation_interval: 200,
+          validation_steps: 100,
+        },
+      },
+    },
+  },
+  local convergence = common.Convergence,
+  local v2_8 = {
+    accelerator: tpus.v2_8,
+    scriptConfig+: {
+      paramsOverride+: {
+        task+: {
+          train_data+: {
+            global_batch_size: 64,
+          },
+        },
+      },
+    },
+  },
+  local val_32 = {
+    scriptConfig+: {
+      paramsOverride+: {
+        task+: {
+          validation_data+: {
+            global_batch_size: 32,
```

Comment:
I would create a common TPU subclass where you compute the validation batch size according to the number of replicas (i.e. `config.accelerator.replicas`). See this example:

https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/7dec2008823f97ec67a4ab444a8975d4aa2e1ad2/tests/tensorflow/r2.8/maskrcnn.libsonnet#L91-L93